Mathlib/Analysis/Normed/Group/Completeness.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2023 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Analysis.Normed.Group.Basic
+import Mathlib.Analysis.SpecificLimits.Basic
+
+/-!
+# Completeness of normed groups
+
+This file includes a completeness criterion for normed additive groups in terms of convergent
+series.
+
+## Main results
+
+* `NormedAddCommGroup.completeSpace_of_summable_implies_tendsto`: A normed additive group is
+complete if an absolutely convergent series converges in the space.
+
+## References
+
+* [bergh_lofstrom_1976] `completeSpace_of_summable_implies_tendsto` and
+  `summable_implies_tendsto_of_complete` correspond to the two directions of Lemma 2.2.1.
+
+## Tags
+
+CompleteSpace, CauchySeq
+-/
+
+open scoped BigOperators Topology
+open Filter Finset
+
+section Metric
+
+variable {α : Type*} [PseudoMetricSpace α]
+
+lemma Metric.exists_subseq_summable_dist_of_cauchySeq (u : ℕ → α) (hu : CauchySeq u) :
+    ∃ f : ℕ → ℕ, StrictMono f ∧ Summable fun i => dist (u (f (i+1))) (u (f i)) := by
+  obtain ⟨f, hf₁, hf₂⟩ :=
+    Metric.exists_subseq_bounded_of_cauchySeq u hu (fun n => (1 / (2:ℝ))^n) (fun n => by positivity)
+  refine ⟨f, hf₁, ?_⟩
+  refine Summable.of_nonneg_of_le (fun n => by positivity) ?_ summable_geometric_two
+  exact fun n => le_of_lt <| hf₂ n (f (n+1)) <| hf₁.monotone (Nat.le_add_right n 1)
+
+end Metric
+
+section Normed
+
+variable {E : Type*} [NormedAddCommGroup E]
+
+/-- A normed additive group is complete if an absolutely convergent series converges in the
+space.  -/
+lemma NormedAddCommGroup.completeSpace_of_summable_implies_tendsto
+    (h : ∀ u : ℕ → E,
+      Summable (‖u ·‖) → ∃ a, Tendsto (fun n => ∑ i in range n, u i) atTop (𝓝 a)) :
+    CompleteSpace E := by
```

Comment:
It seems common to use `imp` as a shorthand for `implies`. Also, it would be nice to make it iff, and the reverse direction shouldn't be hard.

Reply:
The reverse direction is the next lemma, but I also added the iff version at the end.